src/routes/dashboard/components/FilterMenu.svelte

```diff
@@ -0,0 +1,109 @@
+<script lang="ts">
+  import {
+    Column,
+    DatePicker,
+    DatePickerInput,
+    Dropdown,
+    MultiSelect,
+    Row,
+  } from 'carbon-components-svelte'
+  import { timeBetweenDates } from '../helpers/dates'
+
+  export let dates: Date[]
+  export let today: Date
+  export let startDate: Date
+  export let endDate: Date
+  export let channels: string[]
+
+  const dateOptions = { weekday: 'long', year: 'numeric', month: 'long', day: 'numeric' };
+
+  // filter by channel
+  const channelDropdownItems: { id: string; text: string }[] = []
+  let i = 0
+  for (const channel of channels) {
+    channelDropdownItems.push({
+      id: i.toString(),
+      text: channel,
+    })
+    i++
+  }
```

Comment:
there's a nifty trick to use Array.entries and destructure the item and index https://flaviocopes.com/how-to-get-index-in-for-of-loop/

ref https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/entries